update FAQ

ManonLef opened the pull request:
## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->
A few FAQ sections in this document are outdated and not in sync with the current Discord status

## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->
- update information about Thinkful to reflect the latest QA on Discord as edited in April
- add a link to the guide on how to use modmail as added in the `#contact-moderators` channel on discord
- remove help vampire reference and link
- update link on asking questions to lead to the TOP article
- update bootcamp info to reflect current QA on Discord
- 2023-05-23 check and update links 

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.
-->
Closes #XXXXX

## Additional Information
<!-- Any other information about this PR, such as a link to a Discord discussion. -->

If this PR is merged the following needs to be done:
- [ ] Confirm the discord handles are correctly formatted for posting through Dyno/Thor i.e. `<@575252669443211264>` 
- [ ] Update the discord FAQs with Dyno/Thor

If anyone has experience with the above and has the Dyno perms required to do so AND is willing to pick this up, please make yourself known :)

rlmoser99 commented on FAQ.md:
```diff
@@ -93,17 +93,13 @@ Very, but that doesn't mean you can't do it. Learning programming in general is
 **How should I ask questions?**
 
-[Read this]([https://discord.com/channels/505093832157691914/505093832157691916/853784735292325898](https://www.theodinproject.com/how_to_ask)).
+[Read this](https://www.theodinproject.com/how_to_ask).
```
This link's path has been updated to: https://www.theodinproject.com/guides/community/how_to_ask

ManonLef replied:
Thanks for catching that @rlmoser99 
I noticed more links had to be changed so I edited those as well.

rlmoser99 reviewed (approved):
Looks good to me.

ManonLef commented:
@rlmoser99 just to be sure; am I allowed to merge it then?

Would you be willing to:
- confirm the discord modmail handle format is correct for dyno bot and,
- update the FAQ through dyno once merged?

rlmoser99 commented:
@ManonLef Yes, you are good to merge this.

I have updated the links, but I can't update the text about Modmail, because that block of text is at it's max character limit, so I'll need to move text from one message to another, which takes a bit more focus than I have time for right now. But I should have time to do this after work.

rlmoser99 commented:
@ManonLef Just to follow up and let you know these changes have all been made in Dyno

ManonLef commented:
> @ManonLef Just to follow up and let you know these changes have all been made in Dyno

Thank you @rlmoser99 

I do notice other sections and rewording are not in sync yet with the FAQ on discord. In the previous PR, we also changed some link formatting and added a link to the getting hired section for example. 

Is there a way I could format this document to make it easier for you to re-post it with dyno to the #FAQ channel instead of having to update each change manually? 

@01zulfi also proposed using our Odin bot to fetch the md file and post it in the relevant channels. Perhaps that is something we could work on? Especially if it can be made so that it will update automatically or more easily.
